src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -779,7 +777,10 @@ private void ParseLoggerElement(NLogXmlElement loggerElement, IList<LoggingRule>
                 }
             }
 
-            rulesCollection.Add(rule);
+            lock (rulesCollection)
+            {
+                rulesCollection.Add(rule);
```

Comment:
would not expose `AddLoggingRulesSafe` (making it `internal`), a more logical call here? I think i would prefer it (more explit, `CloneLoggingRulesSafe` is alos `internal`)

Reply:
Not possible for the same method is reused for LoggingRules and ChildRules.

Reply:
good one, thanks